kge/job/eval_entity_ranking.py

```diff
@@ -217,8 +219,22 @@ def _evaluate(self):
                 o_in_chunk_mask = (chunk_start <= o) & (o < chunk_end)
                 o_in_chunk = (o[o_in_chunk_mask] - chunk_start).long()
                 s_in_chunk = (s[s_in_chunk_mask] - chunk_start).long()
-                scores_sp[o_in_chunk_mask, o_in_chunk] = o_true_scores[o_in_chunk_mask]
-                scores_po[s_in_chunk_mask, s_in_chunk] = s_true_scores[s_in_chunk_mask]
+
+                # check that scoring is consistent up to configured tolerance
+                # if this is not the case, evaluation metrics may be artificially inflated
+                close_check = torch.allclose(scores_sp[o_in_chunk_mask, o_in_chunk], o_true_scores[o_in_chunk_mask],
+                                             rtol=self.tie_rtol, atol=self.tie_atol)
+                close_check &= torch.allclose(scores_po[s_in_chunk_mask, s_in_chunk], s_true_scores[s_in_chunk_mask],
+                                              rtol=self.tie_rtol, atol=self.tie_atol)
+                if not close_check:
+                    diff_a = torch.abs(scores_sp[o_in_chunk_mask, o_in_chunk] - o_true_scores[o_in_chunk_mask])
+                    diff_b = torch.abs(scores_po[s_in_chunk_mask, s_in_chunk] - s_true_scores[s_in_chunk_mask])
+                    diff_all = torch.cat((diff_a, diff_b))
+                    self.config.log(f"Tie-handling: mean difference between scores was: {diff_all.mean()}.")
+                    self.config.log(f"Tie-handling: max difference between scores was: {diff_all.max()}.")
+                    raise ValueError("Tie-handling tolerances set too low! The same triples scored twice did not fall "
```

Comment:
It's not necessarily an error in the tolerance, could also be the implementation. Perhaps write: "Error in tie handling. A triple was assigned different scores by the spo/sp_/_po scoring implementations during evaluation. Verify model implementation or consider increasing tie-handling tolerances."